sound/soc/sof/ipc.c

```diff
@@ -285,31 +243,29 @@ int sof_ipc_tx_message(struct snd_sof_ipc *ipc, u32 header,
 
 	spin_lock_irq(&sdev->ipc_lock);
 
-	/* get an empty message */
-	msg = msg_get_empty(ipc);
-	if (!msg) {
-		spin_unlock_irq(&sdev->ipc_lock);
-		ret = -EBUSY;
-		goto unlock;
-	}
+	/* initialise the message */
+	msg = &ipc->msg;
 
 	msg->header = header;
 	msg->msg_size = msg_bytes;
 	msg->reply_size = reply_bytes;
-	msg->ipc_complete = false;
 
 	/* attach any data */
 	if (msg_bytes)
 		memcpy(msg->msg_data, msg_data, msg_bytes);
 
-	/* add message to transmit list */
-	list_add_tail(&msg->list, &ipc->tx_list);
-
 	/* send the message if not busy */
-	if (snd_sof_dsp_is_ipc_ready(sdev))
+	if (snd_sof_dsp_is_ipc_ready(sdev)) {
 		ret = ipc_tx_next_msg(sdev->ipc);
```

Comment:
@lyakh can we possibly remove ipc_tx_next_msg and call snd_sof_dsp_send_msg directly here? next_msg made sense with a list but not anymore

Reply:
oops sorry I see you've already done that. github;s ordering of commits is weird.